### PR TITLE
Add Node.js multitenancy guide

### LINKED ISSUE #2703: Add guide on tenant tokens with Node.js
We want to repurpose content from this blog post: https://blog.meilisearch.com/multi-tenancy/

Instead of using and building upon the [tutorial repository](https://github.com/meilisearch/tutorials/tree/main/src/tenant-token-tutorial), we want to extract the instructions about managing tenant tokens with Node.js and build a stand-alone guide.

### PR DESCRIPTION
**Pull request by Strift**
# Pull Request

## Related issue
Fixes #2703 

## What does this PR do?

This PR repurposes the instructions from this [Multi-tenancy guide](https://blog.meilisearch.com/multi-tenancy/) to create a guide focusing on implementing multi-tenancy with Nodejs.

To be completely honest, I'm a bit hesitant about the value this adds compared to the [tenant tokens docs](https://www.meilisearch.com/docs/learn/security/tenant_tokens), so feedback is welcome @guimachiavelli @CaroFG 

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (CaroFG) [inline learn/cookbooks/multitenancy_nodejs.mdx:11]: Not sure if this is really a problem, but it's the exact same sentence we use in the multitenancy and tenant tokens section of the documentation

**Comment by CaroFG** [comment]
Honestly, I don't think this guide adds anything to the documentation. What we explain here is basically the same as what's in the dedicated section of the documentation. I think it could be a bit longer and add something different, like explaining how to connect the back end and front end.

**Comment by Strift** [comment]
Thanks for the feedback @CaroFG, I'll see how I can revisit this.

**Comment by Strift** [comment]
I've updated the guide. I think this is ready for review @CaroFG @guimachiavelli.

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (CaroFG) [inline learn/cookbooks/multitenancy_nodejs.mdx:61]:
```suggestion
You will generate a tenant token that will allow a given patient to search only for their appointments. To achieve this, you will create a tenant token that filters the results based on the patient's ID.
```

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (CaroFG) [inline learn/cookbooks/multitenancy_nodejs.mdx:212]: "and to make searches using the tenant token."
This part of the sentence doesn't flow well in my opinion, it feels disconnected.

```suggestion
In this guide, you saw how to implement secure, multitenant search in a Node.js application. You then created an endpoint to generate tenant tokens for each user. You also built a search interface with InstantSearch to make searches using the tenant token
```

**Review by CaroFG — DISMISSED** [review]

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (guimachiavelli) [inline learn/cookbooks/multitenancy_nodejs.mdx:9]:
```suggestion
This guide will walk you through implementing search in a multitenant Node.js application handling sensitive medical data.
```

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (guimachiavelli) [inline learn/cookbooks/multitenancy_nodejs.mdx:61]:
```suggestion
The first step is generating a tenant token that will allow a given patient to search only for their appointments. To achieve this, you must first create a tenant token that filters results based on the patient's ID.
```

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (guimachiavelli) [inline learn/cookbooks/multitenancy_nodejs.mdx:63]:
```suggestion
Create a `search.js` file and use the following code to generate a tenant token:
```

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (guimachiavelli) [inline learn/cookbooks/multitenancy_nodejs.mdx:100]: I think you don't need this sentence, since the first paragraph in the next section more or less repeats the same thing. Your call, though, feel free to ignore if you disagree.

```suggestion
```

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (guimachiavelli) [inline learn/cookbooks/multitenancy_nodejs.mdx:106]:
```suggestion
Now that you have a tenant token, use it to perform searches. To achieve this, you will need to:
```

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (guimachiavelli) [inline learn/cookbooks/multitenancy_nodejs.mdx:108]:
```suggestion
- On the server: create an endpoint to send the token to your front-end
- On the client: retrieve the token and use it to perform searches
```

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (guimachiavelli) [inline learn/cookbooks/multitenancy_nodejs.mdx:112]:
```suggestion
This guide uses [Express.js](https://expressjs.com/en/starter/installing.html) to create the server. You can install `express` by running:
```

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (guimachiavelli) [inline learn/cookbooks/multitenancy_nodejs.mdx:120]: I think it would be ok if you picked one package manager and removed the others. This way users don't need to make a choice.

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (guimachiavelli) [inline learn/cookbooks/multitenancy_nodejs.mdx:144]: You could possibly move the last sentence ("You will use this endpoint […]") to the beginning of the next sections. Seems more in line with the pacing you set in the rest of the guide.

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (Strift) [reply learn/cookbooks/multitenancy_nodejs.mdx:120]: I feel like this is less and less common in the JS world, and the standard is to remind people that they can use different package managers. I will leave it like this for now, but we can definitely revisit later.

File: learn/cookbooks/multitenancy_nodejs.mdx
Comment (Strift) [reply learn/cookbooks/multitenancy_nodejs.mdx:139]: Thanks for the suggestion!

**Comment by Strift** [comment]
Thanks for the review @guimachiavelli 🙏  I applied most suggestions and fixed conflicts. All green to merge on my end 👌

**Review by guimachiavelli — APPROVED** [review]

**Comment by guimachiavelli** [comment]
Looks good to go for me. I see @CaroFG had a few comments, so I'll wait for her ok before merging

**Comment by CaroFG** [comment]
LGTM!

**Pull request merged**
